src/extended-time-element.js

```diff
@@ -3,21 +3,13 @@
 import {makeFormatter} from './utils'
 
 export default class ExtendedTimeElement extends HTMLElement {
-  _date: ?Date
-
   static get observedAttributes() {
     return ['datetime', 'day', 'format', 'hour', 'minute', 'month', 'second', 'title', 'weekday', 'year']
   }
 
   // Internal: Refresh the time element's formatted date when an attribute changes.
-  //
-  // Returns nothing.
+  // eslint-disable-next-line no-unused-vars
   attributeChangedCallback(attrName: string, oldValue: string, newValue: string) {
-    if (attrName === 'datetime') {
-      const millis = Date.parse(newValue)
-      this._date = isNaN(millis) ? null : new Date(millis)
-    }
```

Comment:
The nice thing about the callback is that we're parsing the date only once and caching its value until it's changed. Depending on how frequently the new getter method is called the date parsing could show up in performance timing graphs, similar to #96.

Reply:
I was curious to see if Date.parse() was really that slow compared to getting from a WeakMap; I have to say [the results are surprising](https://jsperf.com/parsing-date-vs-caching-in-weakmap). WeakMap is not much faster for unhappy paths (I presume because `Date.parse()` knows how to bail out quickly). Of course I'd say given how trivial it is for us to implement a WeakMap here, we should.